src/main/java/com/github/javabdd/BDDDomain.java

```diff
@@ -405,7 +405,7 @@ public BigInteger[] getVarIndices(BDD that) {
      */
     public BigInteger[] getVarIndices(BDD that, int max) {
         BDDVarSet myvarset = set(); // can't use var here, must respect subclass a factory may provide
-        int n = (int)that.satCount(myvarset);
+        int n = that.satCount(myvarset).intValueExact();
```

Comment:
Nice that `intValueExact` throws an exception if `that.satCount(myvarset)` goes out of range for the integer type. Now `n` can no longer become negative due to overflows.